tests/wpt/grouping_formatter.py

```diff
@@ -17,16 +17,22 @@ def __init__(self):
         self.need_to_erase_last_line = False
         self.current_display = ""
         self.running_tests = {}
-        self.last_test_finished = "Running tests..."
         self.test_output = collections.defaultdict(str)
         self.subtest_failures = collections.defaultdict(list)
         self.test_failure_text = ""
         self.tests_with_failing_subtests = []
         self.interactive = os.isatty(sys.stdout.fileno())
 
-        # iTerm2 doesn't support the terminal codes used to erase previous lines,
-        # so only print one line and rely only on backspace characters.
-        self.one_line = os.environ.get("TERM_PROGRAM", "") == "iTerm.app"
+        # TODO(mrobinson, 8313): We need to add support for Windows terminals here.
+        if self.interactive:
+            try:
+                import blessings
+                self.terminal = blessings.Terminal()
+                self.clear_eol = self.terminal.clear_eol
+                self.move_up = self.terminal.move_up
+            except:
```

Comment:
The idea here was that if the import failed, if creating the Terminal failed, or if the ```__getattr__``` call in blessings fails (which <a href="https://github.com/erikrose/blessings/blob/master/blessings/__init__.py#L157">just calls into curses</a>), we should always fall back. If you like I can split the work into two try/except blocks and print an error in the second one.